src/telemetry.ts

```diff
@@ -145,14 +87,21 @@ export class Telemetry {
 			properties = {};
 		}
 
-		// Augment the properties structure with additional common properties before sending
-		Promise.all([this.getUserId(), this.getPlatformInformation()]).then(() => {
-			properties['userId'] = this.userId;
-			properties['distribution'] = (this.platformInformation && this.platformInformation.distribution) ?
-				`${this.platformInformation.distribution.name}, ${this.platformInformation.distribution.version}` : '';
-
+		try {
 			this.reporter.sendTelemetryEvent(eventName, properties, measures);
-		});
+		} catch (telemetryErr) {
+			console.log('Failed to send telemetry event. error: ' + telemetryErr)
```

Comment:
The library already catches and logs errors, so you don't need to do that yourself.

Reply:
removed